src/App.jsx

```diff
@@ -1,248 +1,290 @@
-            <a
-              data-cy="FilterAllUsers"
-              href="#/"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 1
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-              className="is-active"
-            >
-              User 2
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 3
-            </a>
-          </p>
-
-          <div className="panel-block">
-            <p className="control has-icons-left has-icons-right">
-              <input
-                data-cy="SearchField"
-                type="text"
-                className="input"
-                placeholder="Search"
-                value="qwe"
-              />
-
-              <span className="icon is-left">
-                <i className="fas fa-search" aria-hidden="true" />
-              </span>
-
-              <span className="icon is-right">
-                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-                <button
-                  data-cy="ClearButton"
-                  type="button"
-                  className="delete"
-                />
-              </span>
+import cn from 'classnames';
+
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+let areNoneResultsVisible = false;
+// let isAllCategoriesSelected = true;
+// const ALL_CATEGORIES = categoriesFromServer.map(category => category.title);
+
+function getCategoryById(categoryId) {
+  return categories.find(category => category.id === categoryId)
+      || null;
+}
+
+function getUserById(ownerId) {
+  return usersFromServer.find(user => user.id === ownerId)
+      || null;
+}
+
+const categories = categoriesFromServer.map(category => ({
+  ...category,
+  user: getUserById(category.ownerId),
+}));
+
+const products = productsFromServer.map(product => ({
+  ...product,
+  category: getCategoryById(product.categoryId),
+}));
+
+function getFilteredProducts(
+  productsRecieved,
+  { filterOwner,
+    query },
+) {
+  let preparedProducts = [...productsRecieved];
+  const queryOrganised = query.trim().toLowerCase();
+
+  areNoneResultsVisible = false;
+
+  if (filterOwner && filterOwner !== 'All') {
+    preparedProducts = preparedProducts
+      .filter(product => product.category.user.name === filterOwner);
+  }
+
+  if (query) {
+    preparedProducts = preparedProducts
+      .filter(product => product.name.toLowerCase().includes(queryOrganised));
+  }
+
+  // if (filterCategories) {
+  //   // isAllCategoriesSelected = false;
+  //   preparedProducts = preparedProducts
+  //     .filter(product => filterCategories.includes(product.category.title));
+  // }
+
+  if (preparedProducts.length === 0) {
```

Comment:
you can use `preparedProducts.length === 0` for conditional rendering, as you always have up-to-date data you don't need to keep in mind variable that you need to always change